pkg/sql/catalog/descs/factory.go

```diff
@@ -33,8 +37,19 @@ type CollectionFactory struct {
 	spanConfigSplitter spanconfig.Splitter
 	spanConfigLimiter  spanconfig.Limiter
 	defaultMonitor     *mon.BytesMonitor
+	ieFactoryWithTxn   InternalExecutorFactoryWithTxn
 }
 
+// InternalExecutorFactoryWithTxn is used to create an internal executor
+// with associated extra txn state information.
+type InternalExecutorFactoryWithTxn func(
```

Comment:
it feels to me like this should just be a method on the `InternalExecutorFactory` and that this library should describe the interface method. That way, we can avoid the closure in the `server` package.

Reply:
Yeah I thought about that but it can be a bit difficult -- we will have to import `descs` package to `sqlutil` for the collection parameter, but that will again bring dependency loop.

Reply:
I don't think that's true, `sqlutil` doesn't need to include this method in its interface, just the `sql.InternalExecutorFactory` has to implement the method. As in, we define an interface in `descs` which is distinct from the interface in `sqlutil`.

Reply:
Ah, i see what you meant. Yeah, that makes sense, will do!

Reply:
Done.